src/tests/ftest/util/slurm_utils.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/python
+'''
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+
+import os, sys, random, time, subprocess, threading
+import pyslurm
+
+_lock = threading.Lock()
+
+class SlurmFailed(Exception):
+    """ Thrown when something goes wrong with slurm """
+
+def write_slurm_script(path, name, output, nodecount, cmds, sbatch=None):
+    """
+    Generates a script for submitting a job to slurm.
+
+    path      --where to write the script file
+    name      --job name
+    output    --where to put the output (full path)
+    nodecount --number of compute nodes to execute on
+    cmds      --shell commands that are to be executed
+    sbatch    --dictionary containing other less often used parameters to
+                sbatch, e.g. mem:100
+    returns   --the full path of the script
+    """
+
+    if name is None or output is None or nodecount is None or cmds is None:
+        raise SlurmFailed("Bad parameters passed for slurm script.")
+
+    unique = random.randint(1, 100000)
+
+    if not os.path.exists(path):
+        os.makedirs(path)
+    scriptfile = path + '/jobscript' + str(unique) + ".sh"
+
+    f = open(scriptfile, 'w')
```

Comment:
You are OK with this function pushing exceptions up to the caller?

Reply:
Unhandled exception fails the test which is what you want I think.

Reply:
Yup, it should.  I just wanted to confirm that that was your intention.